core/src/main/java/bisq/core/trade/protocol/bsq_swap/tasks/buyer/ProcessBsqSwapFinalizeTxRequest.java

```diff
@@ -39,20 +51,58 @@ public ProcessBsqSwapFinalizeTxRequest(TaskRunner<BsqSwapTrade> taskHandler, Bsq
     @Override
     protected void run() {
         try {
-            runInterceptHook();
-
             BsqSwapFinalizeTxRequest request = checkNotNull((BsqSwapFinalizeTxRequest) protocolModel.getTradeMessage());
             checkNotNull(request);
             Validator.checkTradeId(protocolModel.getOfferId(), request);
 
-            BsqSwapTradePeer tradePeer = protocolModel.getTradePeer();
+            // We will use only the sellers inputs from the tx so we do not verify anything else
+            byte[] tx = request.getTx();
+            Transaction sellersTransaction = protocolModel.getBtcWalletService().getTxFromSerializedTx(tx);
+            List<RawTransactionInput> rawInputs = request.getBtcInputs();
+            checkArgument(!rawInputs.isEmpty(), "Sellers BTC inputs must not be empty");
+
+            int buyersInputSize = Objects.requireNonNull(protocolModel.getInputs()).size();
+            List<TransactionInput> sellersBtcInputs = sellersTransaction.getInputs().stream()
+                    .filter(i -> i.getIndex() >= buyersInputSize)
+                    .collect(Collectors.toList());
+            checkArgument(sellersBtcInputs.size() == rawInputs.size(),
+                    "Number of inputs in tx must match the number of rawInputs");
+
+            boolean hasUnSignedInputs = sellersBtcInputs.stream().anyMatch(i -> i.getScriptSig() == null && !i.hasWitness());
+            checkArgument(!hasUnSignedInputs, "Inputs from tx has unsigned inputs");
 
-            tradePeer.setTx(request.getTx());
-            tradePeer.setInputs(request.getBtcInputs());
+            long sumInputs = rawInputs.stream().mapToLong(rawTransactionInput -> rawTransactionInput.value).sum();
+            int sellersTxSize = BsqSwapCalculation.getTxSize(protocolModel.getTradeWalletService(), rawInputs, request.getBtcChange());
+            long sellersBtcInputAmount = BsqSwapCalculation.getSellersBtcInputValue(trade, sellersTxSize, getSellersTradeFee()).getValue();
+            checkArgument(sumInputs >= sellersBtcInputAmount,
+                    "Sellers BTC input amount do not match our calculated required BTC input amount");
+
+            long change = request.getBtcChange();
+            checkArgument(change == 0 || Restrictions.isAboveDust(Coin.valueOf(change)),
+                    "BTC change must be 0 or above dust");
```

Comment:
Trusting change from peer, better to verify change using `BsqSwapCalculation`

Reply:
Will add verification for that